src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverter.CieXyz.cs

```diff
@@ -13,7 +13,7 @@ public partial class ColorSpaceConverter
 {
     private static readonly HunterLabToCieXyzConverter HunterLabToCieXyzConverter = new();
 
-    private LinearRgbToCieXyzConverter linearRgbToCieXyzConverter;
+    private LinearRgbToCieXyzConverter? linearRgbToCieXyzConverter;
```

Comment:
This is a very strange field, changing it's value for different values of parameters passed to `GetLinearRgbToCieXyzConverter` and it's callers. If there are shared instances, they are not thread-safe. @JimBobSquarePants thoughts? Is it some optimization?

Reply:
ColorSpaceConverter is not threadsafe. 

Whole thing needs a rework tbh. I copied the design, and it doesn't allow easy expansion or optimization.

Reply:
It was an optimization hack. I've replaced it with a concurrent dictionary. Should be safe now.